src/components/grid.jsx

```diff
@@ -388,6 +422,35 @@ class Grid extends React.Component {
       this.props.highlightedSnake
     );
 
+    if (!this.props.highlightedSnake) {
```

Comment:
This whole block _feels_ like it would be pretty prone to performance issues with how many nested loops there are. HOWEVER, I don't really have any evidence to back that up.

Reply:
possibly but this looping is pretty fast, even with large boards.  an alternative might be creating a set of coordinates and searching that set instead, not sure if that might be more efficient